src/engineio/packet.rs

```diff
@@ -41,6 +43,24 @@ const fn u8_to_packet_id(b: u8) -> Result<PacketId> {
     }
 }
 
+/// Data which gets exchanged in a handshake as defined by the server.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct HandshakePacket {
+    pub sid: String,
+    pub upgrades: Vec<String>,
+    #[serde(rename = "pingInterval")]
+    pub ping_interval: u64,
+    #[serde(rename = "pingTimeout")]
+    pub ping_timeout: u64,
+}
+
+impl TryInto<HandshakePacket> for Packet {
+    type Error = Error;
+    fn try_into(self) -> Result<HandshakePacket> {
+        Ok(serde_json::from_slice::<HandshakePacket>(self.data[..].as_ref())?)
```

Comment:
I think we don't need a turbofish `::<>`here anymore, type inference will catch this.

Reply:
done.